docker/supabase/volumes/db/init/05-github-schema.sql

```diff
@@ -0,0 +1,29 @@
+CREATE TABLE IF NOT EXISTS github_users (
+  id serial PRIMARY KEY,
+  username text, -- the user name of the user, known as the `login` field in GitHub API responses
+  user_type text check ( user_type IN ('User', 'Organization') )
+);
+
+CREATE TABLE IF NOT EXISTS github_repositories (
+  id serial PRIMARY KEY,
+  repo_name text,
+  owner integer references github_users ON DELETE CASCADE
+);
+
+-- Although pull requests have different features from
+-- regular issues, they can be treated the same for our purposes
+CREATE TABLE IF NOT EXISTS github_issues (
+  id serial PRIMARY KEY,
```

Comment:
Question: will this give us everything we need to query for the item?  For example, a [pull request needs](https://docs.github.com/en/rest/reference/pulls#get-a-pull-request):

```
GET /repos/{owner}/{repo}/pulls/{pull_number}
```

And an [issues needs](https://docs.github.com/en/rest/reference/issues#get-an-issue):

```
GET /repos/{owner}/{repo}/issues/{issue_number}
```

Given these requirements, one thing I notice is that we need to include whether this is an `issue` or `pull request`, since we have to use `.../issues/...` or `.../pulls/...` in the `GET` request.

Reply:
Well, we can query just fine from the issues endpoint. However, we won't be able to differentiate them, since we are treating PRs are issues (which is technically fine for the API). I will add an extra column that also records the type.